src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -265,6 +265,16 @@ public BoundExpression VisitBinaryOperator(BoundBinaryOperator node, BoundUnaryO
                     case BinaryOperatorKind.BoolAnd:
                         if (loweredRight.ConstantValue == ConstantValue.True) return loweredLeft;
                         if (loweredLeft.ConstantValue == ConstantValue.True) return loweredRight;
+
+                        if (loweredLeft.IsDefaultValue())
```

Comment:
This applies to any argument to boolean And. Nullable or not. 
If one of the sides is a trivial value, we know the result of And, just may need to run the other operand for sideeffects.